cdci_data_analysis/flask_app/dispatcher_query.py

```diff
@@ -526,11 +534,42 @@
 
         logger.warn('-----> set status to %s', status)
 
-        job.write_dataserver_status(
-            status_dictionary_value=status, full_dict=self.par_dic)
+        if self.is_email_to_send_callback(status):
+            try:
+                self.send_email(status,
+                                instrument=instrument_name,
+                                time_request=time_request)
+                self.par_dic['mail_status'] = 'mail sent'
+                job.write_dataserver_status(status_dictionary_value=status, full_dict=self.par_dic)
+            except MailNotSent as e:
+                self.par_dic['mail_status'] = 'mail not sent'
+                job.write_dataserver_status(status_dictionary_value=status, full_dict=self.par_dic)
+        else:
+            job.write_dataserver_status(status_dictionary_value=status, full_dict=self.par_dic)
 
         return status, query_out
 
+    def is_email_to_send_callback(self, status):
+        # get total request duration
+        duration_query = -1
+        if 'time_request' in self.par_dic:
+            time_request = float(self.par_dic['time_request'])
+            duration_query = time_.time() - time_request
+        if self.token:
+            resp = self.validate_token_request_param()
+            if resp is not None:
+                self.logger.warning("query dismissed by token validation")
+                return resp
+            timeout_threshold_mail = tokenHelper.get_token_user_timeout_threshold_mail(self.decoded_token)
+            if timeout_threshold_mail is None:
+                # set it to the a default value, from the configuration
+                timeout_threshold_mail = self.app.config.get('conf').mail_sending_timeout_threshold
+            # in case the request was long and 'done'
+            # or if failed
+            # or when the job was created ('submitted')
+            return (duration_query > timeout_threshold_mail and status == 'done') or status == 'submitted' or status == 'failed'
```

Comment:
I am not sure `submitted` ever happens during callback, except perhaps due to an issue. Generally it will only happen during `run_analysis` request. So email might be sent in both calls.

Reply:
I will include also the case `submitted` for `run_analysis`

Reply:
For my understanding, we want to have an email when the query status becomes `submitted`? This happens here:
 [https://github.com/oda-hub/dispatcher-app/blob/b9191b653203859745117d008cc41de524d2d044/cdci_data_analysis/flask_app/dispatcher_query.py#L1195-L1205](https://github.com/oda-hub/dispatcher-app/blob/b9191b653203859745117d008cc41de524d2d044/cdci_data_analysis/flask_app/dispatcher_query.py#L1195-L1205)

So, afterwards, the user will be receiving updates by a number of `call_back` call?